src/main/java/org/candlepin/resource/CloudRegistrationResource.java

```diff
@@ -160,6 +163,52 @@ else if (version == 2) {
         }
     }
 
+    @Override
+    public List<String> cancelCloudAccountJobs(String cloudAccountId) {
+
+        if (cloudAccountId == null || cloudAccountId.isBlank()) {
+            throw new BadRequestException(this.i18n.tr("Cloud account ID is null or empty"));
+        }
+
+        AsyncJobStatusQueryArguments asyncJobStatusQueryArguments = new AsyncJobStatusQueryArguments();
+        asyncJobStatusQueryArguments.setJobKeys(CloudAccountOrgSetupJob.JOB_KEY)
+            .setJobStates(AsyncJobStatus.JobState.CREATED, AsyncJobStatus.JobState.QUEUED,
+                AsyncJobStatus.JobState.RUNNING, AsyncJobStatus.JobState.SCHEDULED,
+                AsyncJobStatus.JobState.WAITING, AsyncJobStatus.JobState.FAILED_WITH_RETRY);
+        List<AsyncJobStatus> jobs = this.jobManager.findJobs(asyncJobStatusQueryArguments);
+
+        jobs = jobs.stream()
+            .filter(job -> job.getJobArguments().getAsString("cloud_account_id").equals(cloudAccountId))
+            .collect(Collectors.toList());
+
+        jobs.forEach(job -> {
+            jobManager.cancelJob(job.getId());
```

Comment:
There is a good chance that the cancelJob method can throw an `IllegalStateException` with the message "job is already in a terminal state", because between the time the query returns that a job is in QUEUED state, and the time you try to cancel it, the state could have changed to be terminal already. Can we catch those here, check for that exact message, and just continue?

Reply:
Also, a unit test with this use case would be good

Reply:
done